### DISCUSSION TITLE
[OP Stack Deployment] No matching artifact found

### DISCUSSION THREAD

**Discussion by yiming-face** [ethereum-optimism/developers#344] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

after run 
`forge script scripts/Deploy.s.sol:Deploy --sig 'sync()' --rpc-url $L1_RPC_UR`
get this error 
"No matching artifact found"

at the second sync call:  Syncing deployment DisputeGameFactory: contract DisputeGameFactory.0.8.15

foundry tries to call function sync, which is looking for contract DisputeGameFactory.0.8.15 , but this is clearly not a correct contract name

and in my forge-artifacts directory, under DisputeGameFactory.sol, there are two json files:
DisputeGameFactory.0.8.15.json
DisputeGameFactory.0.8.19.json

I think this may affect the "sync" call, 

does anyone get the same issue?

### Logs

  Connected to network with chainid 11155111
  Storing temp deployment data in /home/ubuntu/op-stack/optimism/packages/contracts-bedrock/deployments/getting-started/.deploy
  DeployConfig: reading file /home/ubuntu/op-stack/optimism/packages/contracts-bedrock/deploy-config/getting-started.json
  Deploying from Deploy
  Deployment context: getting-started
  Syncing 27 deployments
  Using deployment artifact /home/ubuntu/op-stack/optimism/packages/contracts-bedrock/broadcast/Deploy.s.sol/11155111/run-latest.json
  Syncing deployment AddressManager: contract AddressManager
  Syncing deployment DisputeGameFactory: contract DisputeGameFactory.0.8.15

**Comment by Chomtana** [comment]
Hi, please deploy contracts from `op-contracts/v1.3.0` branch and switch back to `v1.7.2` branch for running op-node, op-batcher, and op-proposer

```
cd optimism
git checkout op-contracts/v1.3.0

... Deploy contract and init genesis commands ...

git checkout v1.7.2

... Start OP Stack chain commands ...
```

Sorry for misleading docs.

**Reply by yiming-face** [reply]
thanks for the reply, I will try it soon. 
I am wondering why there is no such question, seems everyone knows how to do it correctly?

**Reply by Chomtana** [reply]
There is a related discussion at https://github.com/ethereum-optimism/developers/discussions/329

**Reply by hardXcoder** [reply]
<img width="779" alt="image" src="https://github.com/ethereum-optimism/developers/assets/17654427/7795ca00-8814-4e9e-b142-3e2e30fa6790">
Facing this problem, after switching to the branch you mentioned and then running the below command 

forge script scripts/Deploy.s.sol:Deploy --private-key $GS_ADMIN_PRIVATE_KEY --broadcast --rpc-url $L1_RPC_URL --slow

**Reply by hardXcoder** [reply]
@Chomtana

**Reply by Chomtana** [reply]
This is weird! Is your foundry the latest version?

Let try `foundryup`

**Reply by hardXcoder** [reply]
I change the visibility of the functions in the contracts mentioned in the error. Removed the "view" modifier, then it compiles. Is it the right way ?

**Reply by yiming-face** [reply]
I have the same err as @hardXcoder after switch to branch op-contracts/v1.3.0

and here is my foundryup print out...

foundryup: installed - forge 0.2.0 (72bc4f4 2024-04-08T00:16:33.234657619Z)
foundryup: installed - cast 0.2.0 (72bc4f4 2024-04-08T00:16:33.255574701Z)
foundryup: installed - anvil 0.2.0 (72bc4f4 2024-04-08T00:16:33.272515490Z)
foundryup: installed - chisel 0.2.0 (72bc4f4 2024-04-08T00:16:33.271907196Z)
foundryup: done!

**Reply by yiming-face** [reply]
hi, @hardXcoder , after you changed visibility, does the following steps work well?

**Reply by gosunuts** [reply]
It still fail after switch to op-contracts/v1.3.0. cause of failure seems to be latest release of foundry.
set specific version when install foundry. below is example
https://github.com/gokch/optimism_builder/blob/main/init/Dockerfile#L21

**Reply by Chomtana** [reply, marked as answer]
Let's use the branch `tutorials/chain` as pointed out in the Dockerfile you have sent. Thank you @gokch
